pkg/didcomm/crypto/legacy/authcrypt/authcrypt_test.go

```diff
@@ -0,0 +1,198 @@
+	"golang.org/x/crypto/nacl/box"
+	"golang.org/x/crypto/nacl/sign"
+)
+
+func TestEncrypt(t *testing.T) {
+
+	/*
+			Note: box.GenerateKey() generates a curve25519 keypair, not an Ed25519 keypair.
+			I think we should be generating Ed25519 keypairs since they are converted into
+				curve25519 keypairs for processing.
+
+		TODO: use Ed25519 keys, convert to curve25519 internally, encode the Ed25519 form for message packing
+	*/
+
+	t.Run("Success test case: given keys, generate envelope", func(t *testing.T) {
+
+		senderKey := getB58EdKey(
+			"Bxp2KpXeh6RgXXRVGRQUskT9qT35aSSz1JvdbMUcB2Yc",
+			"2QqgiHtrUtDPpfoZG2C3Qi8a1MbLQuTZaaScu5LzQbUCkw5YnXngKLMJ8VuPgoN3Piqt1PBUACVd6uQRmtayZp2x")
+
+		print("sendPK: ", base58.Encode(senderKey.pub[:]), "\n")
+		print("sendSK: ", base58.Encode(senderKey.priv[:]), "\n")
+
+		recipientKey := getB58EdKey(
+			"9ZeipG91uMRDkMbqgkJK2Fq59CoWwfeJx2e5Q543mU5Q",
+			"23Y2dbcT78KEV1T7niUAuf83J8Zta11FG88n6y6pWgZY35nWrhyxGcqCJV5ddHveRZecrhwku77ik7gPSLaXnJXt")
+
+		crypter, e := New(senderKey, []*publicEd25519{recipientKey.pub}, C20P)
+		require.NoError(t, e)
+		require.NotEmpty(t, crypter)
+		enc, e := crypter.Encrypt([]byte("lorem ipsum dolor sit amet"))
+		require.NoError(t, e)
+		require.NotEmpty(t, enc)
+
+		printPythonData(recipientKey.pub[:], recipientKey.priv[:], enc)
+	})
+
+	t.Run("Generate keys, generate envelope", func(t *testing.T) {
+		senderKey, err := randEdKeyPair(rand.Reader)
+		require.NoError(t, err)
+		recipientKey, err := randEdKeyPair(rand.Reader)
+		require.NoError(t, err)
+
+		crypter, e := New(*senderKey, []*publicEd25519{recipientKey.pub}, C20P)
+		require.NoError(t, e)
+		require.NotEmpty(t, crypter)
+		enc, e := crypter.Encrypt([]byte("lorem ipsum dolor sit amet"))
+		require.NoError(t, e)
+		require.NotEmpty(t, enc)
+
+		printPythonData(recipientKey.pub[:], recipientKey.priv[:], enc)
+	})
+
+	t.Run("Generate testcase with multiple recipients", func(t *testing.T) {
+		senderKey, err := randEdKeyPair(rand.Reader)
+		require.NoError(t, err)
+		rec1Key, err := randEdKeyPair(rand.Reader)
+		require.NoError(t, err)
+		rec2Key, err := randEdKeyPair(rand.Reader)
+		require.NoError(t, err)
+		rec3Key, err := randEdKeyPair(rand.Reader)
+		require.NoError(t, err)
+		rec4Key, err := randEdKeyPair(rand.Reader)
+		require.NoError(t, err)
+
+		recipientKeys := []*publicEd25519{
+			rec1Key.pub,
+			rec2Key.pub,
+			rec3Key.pub,
+			rec4Key.pub,
+		}
+
+		crypter, e := New(*senderKey, recipientKeys, C20P)
+		require.NoError(t, e)
+		require.NotEmpty(t, crypter)
+		enc, e := crypter.Encrypt([]byte("The quick brown fox jumped over the lazy dog"))
+		require.NoError(t, e)
+		require.NotEmpty(t, enc)
+
+		printPythonData(rec3Key.pub[:], rec3Key.priv[:], enc)
+	})
+}
+
+func TestSodiumBoxSeal(t *testing.T) {
+	var err error
+
+	recipient1Key, err := randCurveKeyPair(rand.Reader)
+	require.NoError(t, err)
+
+	t.Run("Generate a box_seal message to compare to ACA-Py:", func(t *testing.T) {
```

Comment:
where is the comparison to ACA-Py's output?

Reply:
Randomness (for nonces, CEKs) keeps me from generating encrypted envelopes in ACA-Py to compare against my own output. Right now I generate test data which can be pasted into a python test script to verify using ACA-Py decrypt, but once I have decrypt implemented I can create test data in ACA-Py that can be part of test cases here, testing decrypt, and I can also have test cases for my encrypt against my decrypt.

In essence, verification of correctness needs to happen on the decrypt side. I could work on making encrypt take a Reader as parameter for the randomness source, with the wallet being relied on to submit a CSPRNG.

Reply:
My above concern should be solved now. You should soon see more tests like what you want to see.